sdk/search/Azure.Search.Documents/src/Batching/Publisher.cs

```diff
@@ -422,12 +422,14 @@ private async Task PublishAsync(bool flush, CancellationToken cancellationToken)
             // Returns whether the batch is now full.
             bool FillBatchFromQueue(List<PublisherAction<T>> batch, Queue< PublisherAction<T>> queue)
             {
-                // TODO: Consider tracking the keys in the batch and requiring
-                // them to be unique to avoid error alignment problems
+                HashSet<string> documentIdentifiers = new HashSet<string>(StringComparer.Ordinal);
```

Comment:
Are keys case-sensitive? I'm not actually sure. @brjohnstmsft? I worry about coupling with this behavior as well, but I suppose we don't have a choice. As long as it's documented, we should be fine.

Reply:
Yes, document keys are case-sensitive.

Reply:
I didn't find a documentation mentioning this explicitly. So, I chose case-sensitive to keep the check stricter.

Reply:
@brjohnstmsft is that documented anywhere? Perhaps it should be. I'm not crazy about the idea of taking a dependency on undocumented behavior. At least with a case-sensitive comparer, the service will still err appropriately.

Reply:
I can't find it mentioned anywhere -- good catch. Please open an issue on azure-docs.

Don't worry about taking a dependency on this behaviour though -- it's not going to change.